docs/datamodel/build-a-datamodel.rst

```diff
@@ -0,0 +1,627 @@
+* Mapping external representations to/from the internal representation via
+  loaders and serializers.
+
+You can build data models that are both custom to your exact needs, or you can
+build data models that follow standard metadata formats such as Dublin Core,
+DataCite or MARC21. In fact, a data model does not put any restrictions on what
+you can store, except that a record must be stored internally as JSON.
+
+You can build data models for classic digital repository use cases such as
+bibliographic and author records, but Invenio is in no way limited to these
+classic use cases, and you could as well build your geographical research
+database on top of Invenio.
+
+First steps
+-----------
+First of all, make sure you have followed the :ref:`quickstart`, to ensure
+you have scaffolded an initial *Invenio instance* and *a data model package*.
+
+You should see a directory structure similar to the one below in the newly
+scaffolded data model package:
+
+.. code-block:: shell
+
+    |-- ...
+    |-- docs
+    |   |-- ...
+    |-- my_datamodel
+    |   |-- config.py
+    |   |-- jsonschemas/
+    |   |-- loaders/
+    |   |-- mappings/
+    |   |-- marshmallow/
+    |   |-- serializers/
+    |   |-- static/
+    |   |-- templates/
+    |   `-- ...
+    |-- setup.py
+    `-- tests
+        |-- ...
+
+Building a data model involves the following tasks:
+
+- Define a JSONSchema.
+- Define an Elasticsearch mapping.
+- Define a Marshmallow schema.
+- Define serializers.
+- Define loaders.
+- Define templates.
+- Choose a persistent identifier scheme.
+- Configure the REST API and UI.
+
+Define a JSONSchema
+-------------------
+Internally records are stored as JSON, and in order to validate the structure of
+the stored JSON you must write a `JSONSchema <http://json-schema.org>`_.
+
+The scaffolded data model package includes an example of a simple JSONSchema,
+that you can use to get a feeling of what a JSONSchema looks like.
+
+.. code-block:: shell
+
+    |-- my_datamodel
+    |   |-- jsonschemas
+    |   |   |-- __init__.py
+    |   |   `-- records
+    |   |       `-- record-v1.0.0.json
+
+
+In ``record-v1.0.0.json`` you should see something like:
+
+.. code-block:: json
+
+    {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+        "id": "https://localhost/schemas/records/record-v1.0.0.json",
+        "type": "object",
+        "properties": {
+            "title": {
+            "description": "Record title.",
+            "type": "string"
+            },
+        }
+    }
+
+**Example record**
+
+An example record that validates against this schema could look like:
+
+.. code-block:: json
+
+    {
+        "$schema": "https://localhost/schemas/records/record-v1.0.0.json",
+        "title": "My record"
+    }
+
+Note, that the ``$schema`` key points to the JSONSchema that the record should
+be validated against.
+
+**Discovery of schemas**
+
+Invenio is using standard Python entry points to discover your data model
```

Comment:
What about linking to [entry points docs](https://setuptools.readthedocs.io/en/latest/pkg_resources.html#entry-points)?